crates/ironrdp-acceptor/src/credssp.rs

```diff
@@ -122,19 +149,19 @@ impl<'a> CredsspSequence<'a> {
         }
     }
 
-    pub fn process_ts_request(&mut self, request: TsRequest) -> Result<ServerState, Box<ServerError>> {
-        Ok(self.server.process(request)?)
+    pub fn process_ts_request(&mut self, request: TsRequest) -> CredsspProcessGenerator<'_> {
+        self.server.process(request)
     }
 
     pub fn handle_process_result(
         &mut self,
-        result: Result<ServerState, Box<ServerError>>,
+        result: Result<ServerState, ServerError>,
         output: &mut WriteBuf,
     ) -> ConnectorResult<Written> {
         let (ts_request, next_state) = match result {
-            Ok(ServerState::ReplyNeeded(ts_request)) => (Some(ts_request), CredsspState::Ongoing),
+            Ok(ServerState::ReplyNeeded(ts_request)) => (Some(Box::new(ts_request)), CredsspState::Ongoing),
             Ok(ServerState::Finished(_id)) => (None, CredsspState::Finished),
-            Err(err) => (Some(err.ts_request), CredsspState::ServerError(err.error)),
+            Err(err) => (err.ts_request, CredsspState::ServerError(err.error)),
```

Comment:
suggestion: Instead of boxing the TS request in the success path, what about unboxing the TS request in the error path?